kv/replica_slice.go

```diff
@@ -135,3 +139,21 @@ func (rs replicaSlice) MoveToFront(i int) {
 	copy(rs[1:i+1], rs[0:i])
 	rs[0] = front
 }
+
+func (rs replicaSlice) SortRandomOrder(i int) {
+	l := len(rs) - 1
+	if i > l {
+		log.Infof("out of bound index")
+	}
+
+	firstNotInOrder := i
+	rndLen := l - i
+	rndOrder := rand.Perm(rndLen)
+	rsRand := rs
```

Comment:
The variable names in this method are confusing. `rsRand` doesn't add anything; you can simply use `rs` (it works even though `rs` is a value and not a pointer because the writes pass through to the underlying array). Use the name `firstNotInOrder` for the parameter of the function instead of passing in `i`, renaming it to `firstNotInOrder`, and then reusing `i` for something else below. Rename `l` to `last` to make it clear that it's the index of the last element of the array instead of the array's length (or get rid of the variable completely by using `if firstNotInOrder >= len(rs)` for the bounds check and setting `rndLen = len(rs) - i - 1`